shetland/core/src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java

```diff
@@ -63,106 +83,93 @@ public String getBaseURI() {
-
-        private final String name;
-
-        QueryOptionPredicate(String name) {
-            this.name = name;
-        }
-
-        @Override
-        public boolean test(FilterClause input) {
-            if (ODataConstants.QueryOptions.COUNT.equalsIgnoreCase(name) && input instanceof CountOption) {
-                return true;
-            } else if (ODataConstants.QueryOptions.ORDERBY.equalsIgnoreCase(name) && input instanceof OrderByOption) {
-                return true;
-            } else if (ODataConstants.QueryOptions.SELECT.equalsIgnoreCase(name) && input instanceof SelectOption) {
-                return true;
-            } else if (ODataConstants.QueryOptions.EXPAND.equalsIgnoreCase(name) && input instanceof ExpandOption) {
-                return true;
-            } else if (ODataConstants.QueryOptions.SKIP.equalsIgnoreCase(name) && input instanceof SkipTopFilter) {
-                return ((SkipTopFilter) input).getOperator().equals(FilterConstants.SkipTopOperator.Skip);
-            } else if (ODataConstants.QueryOptions.TOP.equalsIgnoreCase(name) && input instanceof SkipTopFilter) {
-                return ((SkipTopFilter) input).getOperator().equals(FilterConstants.SkipTopOperator.Top);
-            } else if (ODataConstants.QueryOptions.FILTER.equalsIgnoreCase(name) && input instanceof Filter) {
-                return true;
-            }
+    // BaseURI is not always set -> we need to compare each option individually
+    @Override
+    public boolean equals(Object o) {
+        if (!(o instanceof QueryOptions)) {
             return false;
         }
+        QueryOptions obj = (QueryOptions) o;
+        return obj.hasCountOption() == this.hasCountOption()
+                && obj.hasOrderByOption() == this.hasOrderByOption()
+                && obj.hasSelectOption() == this.hasSelectOption()
+                && obj.hasExpandOption() == this.hasExpandOption()
+                && obj.hasSkipOption() == this.hasSkipOption()
+                && obj.hasTopOption() == this.hasTopOption()
+                && obj.hasFilterOption() == this.hasFilterOption()
+                && (obj.hasCountOption()) ? obj.getCountOption().equals(this.getCountOption()) :
+                (obj.hasOrderByOption()) ? obj.getOrderByOption().equals(this.getOrderByOption()) :
+                        (obj.hasSelectOption()) ? obj.getSelectOption().equals(this.getSelectOption()) :
+                                (obj.hasExpandOption()) ? obj.getExpandOption().equals(this.getExpandOption()) :
+                                        (obj.hasSkipOption()) ? obj.getSkipOption().equals(this.getSkipOption()) :
+                                                (obj.hasTopOption()) ? obj.getTopOption().equals(this.getTopOption()) :
+                                                        (!obj.hasFilterOption()) ||
+                                                                obj.getFilterOption().equals(this.getFilterOption());
```

Comment:
No... Just no...

Reply:
I hate it as well - but i needed to satisfy checkstyle as it did not like my previous syntax

previous code:
```
&& obj.hasCountOption() ? obj.getCountOption().equals(this.getCountOption()) : true
```
and checkstyle error:
```
[ERROR] src/main/java/org/n52/shetland/oasis/odata/query/option/QueryOptions.java:[167,17] (coding) SimplifyBooleanExpression: Expression can be simplified.
```

I will look into a nicer solution :+1:

Reply:
done :+1: